test/LongVectorLowering/add.ll

```diff
@@ -0,0 +1,123 @@
+; RUN: clspv-opt --LongVectorLowering %s -o %t
+; RUN: FileCheck %s < %t
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+; This test essentially does:
+;   store(ptr[2], load(ptr[0]) + load(ptr[1]))
+; where load and store operate on <8 x i32>.
+
+define spir_func void @test(i32 addrspace(1)* %ptr) {
+entry:
+  %0 = load i32, i32 addrspace(1)* %ptr, align 4
```

Comment:
You could remove a lot of test code by changing to:
```
define spir_func void @test(<4 x i32>* addrspace(1)* %ptr) {
entry:
  %0 = load <4 x i32>, <4 x i32>* addrspace(1)* ptr, align 16
  %1 = getelementptr <4 x i32>, <4 x i32>* %ptr, i32 1
  %2 = load <4 x i32>, <4 x i32>* addrspace(1)* %1, align 16
  %3 = shufflevector <4 x i32> %0, <4 x i32> %2, <8 x i32> <i32 0, i32 1, i32 2, i32 3, i32 4, i32 5, i32 6, i32 7>
  ; Repeat with for next element. Shuffle in %7
  %add = add <8 x i32> %3, %7
 ; Shuffle into two int4s and store them.
}
```

Reply:
That's a nicer approach indeed, I'll experiment with that.